BatchApex/BatchLeadTest.cls

```diff
@@ -0,0 +1,122 @@
+/**
+ * This class contains unit tests for validating the behavior of Apex classes
+ * and triggers.
+ *
+ * Unit tests are class methods that verify whether a particular piece
+ * of code is working properly. Unit test methods take no arguments,
+ * commit no data to the database, and are flagged with the testMethod
+ * keyword in the method definition.
+ *
+ * All test methods in an organization are executed whenever Apex code is deployed
+ * to a production organization to confirm correctness, ensure code
+ * coverage, and prevent regressions. All Apex classes are
+ * required to have at least 75% code coverage in order to be deployed
+ * to a production organization. In addition, all triggers must have some code coverage.
+ * 
+ * The @isTest class annotation indicates this class only contains test
+ * methods. Classes defined with the @isTest annotation do not count against
+ * the organization size limit for all Apex scripts.
+ *
+ * See the Apex Language Reference for more information about Testing and Code Coverage.
+ */
+ 
+@isTest
+private class BatchLeadTest {
+    static List<Lead> leadList = new List<Lead>(); 
+    static List<Lead> duplicateLeadList = new List<Lead>(); 
+    static Lead  leadRecord;
+    static Integer i = 0;
+    
+    /**
+     * Create bulk data for lead  
+     */
+    static List<Lead> createLeadData() {
+        
+        // Creation of 200 lead records
+        for(Integer j = 0; j < 200; j++) {
+            leadRecord = new Lead(LastName = 'TestLead', Company = 'testCompanyLead',
+                Status = 'Open - Not Contacted');
+         leadList.add(leadRecord);
+        }
+        return leadList;
+    }
+    
+    /**
+     * Check the lead duplication with same records 
+     */
+    static testMethod void testDuplicateRecordsOfLead() {
+        List <Lead> leadInstance = createLeadData();
```

Comment:
It should be List<Lead> and not List <Lead>
Also why such name? leadInstance should be if we have a single lead. Now we have a list do it should be something like leadList